Confusing error message when trying to use engine type the client was not generated with

When generating the client for the library engine, trying to use the binary engine like this:

```
PRISMA_CLIENT_ENGINE_TYPE=binary ts-node index.ts
```

produces the following error:

```
PrismaClientValidationError: Invalid client engine type, please use `library` or `binary`
```

It should only list the engine types this generated client supports.

Yeah ,can anyone help on this? I am getting like this
`⎔ Starting local server...
X [ERROR] PrismaClientValidationError: Invalid client engine type, please use `library` or binary
`
`  {
    clientVersion: '5.10.2'
  }
`

when i try this it showing this error
X [ERROR] Error: PrismaClient is not configured to run in Cloudflare Workers. In order to run Prisma Client on edge runtime, either:

--- a/packages/client/src/runtime/core/init/getEngineInstance.ts
+++ b/packages/client/src/runtime/core/init/getEngineInstance.ts
@@ -85,9 +85,13 @@ export function getEngineInstance({ copyEngine = true }: GetPrismaClientConfig, 
   else if (binaryEngineConfigured && TARGET_BUILD_TYPE === 'binary') return new BinaryEngine(engineConfig)
   else if (accelerateConfigured && TARGET_BUILD_TYPE === 'wasm') return new AccelerateEngine(engineConfig)
   else if (clientEngineConfigured && TARGET_BUILD_TYPE === 'client') return new ClientEngine(engineConfig)
-
+  // reasonable fallbacks in case the conditions above aren't met, we should still try the correct engine
+  else if (TARGET_BUILD_TYPE === 'edge') return new AccelerateEngine(engineConfig)
+  else if (TARGET_BUILD_TYPE === 'library') return new LibraryEngine(engineConfig)
+  else if (TARGET_BUILD_TYPE === 'binary') return new BinaryEngine(engineConfig)
+  else if (TARGET_BUILD_TYPE === 'client') return new ClientEngine(engineConfig)
   // if either accelerate or wasm library could not be loaded for some reason, we throw an error
-  if (TARGET_BUILD_TYPE === 'wasm') {
+  else if (TARGET_BUILD_TYPE === 'wasm') {
     const message = [
       `PrismaClient failed to initialize because it wasn't configured to run in this environment (${
         getRuntime().prettyName
@@ -102,7 +106,5 @@ export function getEngineInstance({ copyEngine = true }: GetPrismaClientConfig, 
     })
   }
 
-  throw new PrismaClientValidationError('Invalid client engine type, please use `library` or `binary`', {
-    clientVersion: engineConfig.clientVersion,
-  })
+  TARGET_BUILD_TYPE satisfies never
 }

--- a/packages/client/src/runtime/core/init/getEngineInstance.ts
+++ b/packages/client/src/runtime/core/init/getEngineInstance.ts
@@ -86,7 +86,7 @@ export function getEngineInstance({ copyEngine = true }: GetPrismaClientConfig, 
   else if (accelerateConfigured && TARGET_BUILD_TYPE === 'wasm') return new AccelerateEngine(engineConfig)
   else if (clientEngineConfigured && TARGET_BUILD_TYPE === 'client') return new ClientEngine(engineConfig)
   // reasonable fallbacks in case the conditions above aren't met, we should still try the correct engine
-  else if (TARGET_BUILD_TYPE === 'edge') return new AccelerateEngine(engineConfig)
+  else if (TARGET_BUILD_TYPE === 'edge') return new DataProxyEngine(engineConfig)
   else if (TARGET_BUILD_TYPE === 'library') return new LibraryEngine(engineConfig)
   else if (TARGET_BUILD_TYPE === 'binary') return new BinaryEngine(engineConfig)
   else if (TARGET_BUILD_TYPE === 'client') return new ClientEngine(engineConfig)

--- a/packages/client/src/runtime/core/init/getEngineInstance.ts
+++ b/packages/client/src/runtime/core/init/getEngineInstance.ts
@@ -106,5 +106,5 @@ export function getEngineInstance({ copyEngine = true }: GetPrismaClientConfig, 
     })
   }
 
-  TARGET_BUILD_TYPE satisfies never
+  return TARGET_BUILD_TYPE satisfies never
 }

